h3/h3.py

```diff
@@ -628,10 +628,10 @@ def hex_ranges(h3_address_list, ring_size):
             # divided by 6 ceiled.
             ring_index = int(
                 math.floor(
-                    (1 + math.sqrt(1 + 8 * math.ceil(j / 6.0))) / 2)) - 1
+                    (1 + math.sqrt(1 + 8 * max(math.ceil(j / 6.0 - 1), -0.1))) / 2))
```

Comment:
@dfellis Did you see my revised alternative?

*Outside the current `j` loop:
```
ring_index = 0
ring_end = 0
```
*Inside the current `j` loop:
```
if j > ring_end:
  ring_index = ring_index + 1
  ring_end = ring_end + 6 * ring_index
```
It's simpler, more legible, and much faster because it's not doing a `sqrt` op for every single hexagon.